server/neptune/workflows/internal/pr/runner.go

```diff
@@ -90,15 +112,36 @@ func (r *Runner) Run(ctx workflow.Context) error {
 		c.Receive(ctx, &request)
 	})
 
+	onShutdownPollTick := func(f workflow.Future) {
+		action = onShutdownPoll
+	}
+	_, _ = s.AddTimeout(ctx, r.ShutdownPollTick, onShutdownPollTick)
+
 	for {
 		s.Select(ctx)
 		switch action {
 		case onNewRevision:
 			r.onNewRevision(ctx, prRevision)
+			inactivityTimeoutCancel()
+			inactivityTimeoutCancel, _ = s.AddTimeout(ctx, r.InactivityTimeout, onInactivityTimeout)
+		case onCancel:
+			continue
+		case onTimeout: // TODO: send message to PR stating atlantis deleted state due to inactivity and to rerun to trigger atlantis workflow
+			workflow.GetLogger(ctx).Info("workflow timed out, shutting down")
+			r.cancel()
+			return nil
 		case onShutdown:
 			workflow.GetLogger(ctx).Info("received shutdown signal")
 			r.cancel()
 			return nil
+		case onShutdownPoll:
```

Comment:
For simplicity and resiliency i would just combine the two cases:  If you get a signal just retrieve the PR state and shutdown. In that way we're also resilient to bad signals/events.  (ie. this workflow getting called by mistake with a shutdown signal if a PR hasn't been closed yet)